### ISSUE TITLE
all: ability to provide config for checker

### ISSUE THREAD

**Issue by cristaloleg** [go-critic/go-critic#332]
- Easily pass params to linter(possibly via `BindContext` as separate param)
- Must be a JSON file
- Nice to have a command flag to pass JSON

**Comment by Quasilyte** [comment]
I would like to add: would be great to have JSON that permits comments. Supporting this is like 20 lines of code, but makes config far more maintainable. This is the biggest issue with JSON - no comments with default encoders and decoders (at least for me).

**Comment by cristaloleg** [comment]
```json
{
   "_key": "documentation",
   "key": "value",
}
```
WDYT ?

**Comment by Quasilyte** [comment]
We may want checker to set it's defaults inside `Init()` function.
And it's probably a good idea to have per-checker map, so checker has it's own namespace to mess with.
For example:
```go
func (c *xyzChecker) Init() {
  c.ctx.Config["limit"] = 10
  c.ctx.Config["whitelist"] = "a,b" // Just to demonstrate that it's interface{}
}
// and we want accessors like c.ctx.Config.Int("limit") and
// c.ctx.Config.String("whitelist"), I guess.
```
This way, no checker name dependency, we can rename them as we like.
Checker can't override other checker keys.
Explicit place to set defaults without additional magic.

**Comment by Quasilyte** [comment]
> WDYT ?

Not sure I get it.

**Comment by Quasilyte** [comment]
@cristaloleg what do you think about using `Init` with assignments of default values and overriding them with user-specified values inside checker construction?
(See my comment above.)

**Comment by Quasilyte** [comment]
We can even compare user-specified value with default value type and signal error in case it mismatched.

**Comment by cristaloleg** [comment]
Fixed by #422
